### PR TITLE
Add translatable SMS registration welcome messages

### PR DESCRIPTION
**Pull request by gcapalbo**
@biyeun code review buddy

### CODE REVIEW

File: corehq/apps/sms/forms.py [hunk 1]
```diff
@@ -69,6 +69,18 @@
 TIME_AFTER = "AFTER"
 TIME_BETWEEN = "BETWEEN"
 
+WELCOME_RECIPIENT_NONE = 'NONE'
+WELCOME_RECIPIENT_CASE = 'CASE'
+WELCOME_RECIPIENT_MOBILE_WORKER = 'MOBILE_WORKER'
+WELCOME_RECIPIENT_ALL = 'ALL'
+
+WELCOME_RECIPIENT_CHOICES = (
+    (WELCOME_RECIPIENT_NONE, ugettext_lazy('Nobody')),
```
Comment (biyeun) [inline corehq/apps/sms/forms.py]: @esoergel were we going to move away from `ugettext_lazy` in favor of being explicit about `ugettext_noop` and `uggettext`?

File: corehq/apps/sms/views.py [hunk 2]
```diff
@@ -1552,6 +1554,15 @@ def form(self):
                     domain_obj.sms_case_registration_user_id,
                 "sms_mobile_worker_registration_enabled":
                     enabled_disabled(domain_obj.sms_mobile_worker_registration_enabled),
+                "registration_welcome_message":
+                    (WELCOME_RECIPIENT_ALL
+                     if (domain_obj.enable_registration_welcome_sms_for_case and
```
Comment (nickpell) [inline corehq/apps/sms/views.py]: is there a way to display/format this so the logic is clearer?

File: corehq/apps/sms/forms.py (hunk 1 above)
Comment (gcapalbo) [inline corehq/apps/sms/forms.py]: my feeling is that all three are ok to use, and all three can get you into trouble if not used properly

File: corehq/apps/sms/views.py (hunk 2 above)
Comment (gcapalbo) [inline corehq/apps/sms/views.py]: @nickpell updated

File: corehq/apps/sms/forms.py (hunk 1 above)
Comment (biyeun) [inline corehq/apps/sms/forms.py]: ah, I think if it isn't converted to json then it's fine

**Comment by biyeun** [comment]
waiting for build, then ok to merge by me

File: corehq/apps/sms/views.py (hunk 2 above)
Comment (nickpell) [inline corehq/apps/sms/views.py]: thanks!

File: corehq/apps/sms/forms.py (hunk 1 above)
Comment (esoergel) [inline corehq/apps/sms/forms.py]: Here are some specifics about the differences: https://commcare-hq.readthedocs.org/en/latest/translations.html

**Comment by nickpell** [comment]
merging now for early deploy

**Pull request merged**